### PR TITLE
Add eth_api gasPriceIn3Gears

### PR DESCRIPTION
**Pull request by LeoGuo621**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [x] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/rpc/namespaces/eth/api.go [hunk 1]
```diff
@@ -269,6 +270,40 @@ func (api *PublicEthereumAPI) GasPrice() *hexutil.Big {
...
+		}
+	}
+	// safe low GP = average GP * 0.5, but it will not be less than the minimal GP.
+	safeGp := new(big.Int).Quo(avgGP, big.NewInt(2))
+	if safeGp.Cmp(gasprice.MinPrice) == -1 {
+		safeGp.Set(gasprice.MinPrice)
+	}
+	// fastest GP = average GP * 1.5, but it will not be greater than the max GP.
+	fastestGp := new(big.Int).Add(avgGP, new(big.Int).Quo(avgGP, big.NewInt(2)))
+	if fastestGp.Cmp(gasprice.MaxPrice) == 1 {
```
Comment (zjg555543) [inline app/rpc/namespaces/eth/api.go:307]: reset avgGP = （min + max）/ 2 again?

File: app/rpc/tests/rpc_test.go [hunk 2]
```diff
@@ -389,6 +392,25 @@ func (suite *RPCTestSuite) TestEth_GasPrice() {
 	suite.Require().Equal(mgp.Amount.BigInt(), gasPrice.ToInt())
 }
 
+func (suite *RPCTestSuite) TestEth_GasPriceIn3Gears() {
+	rpcRes, err := CallWithError(suite.addr, "eth_gasPriceIn3Gears", nil)
```
Comment (zjg555543) [inline app/rpc/tests/rpc_test.go:396]: may be this ut can't cover all case

File: app/rpc/namespaces/eth/api.go [hunk 3]
```diff
@@ -269,6 +270,40 @@ func (api *PublicEthereumAPI) GasPrice() *hexutil.Big {
...
+	if appconfig.GetOecConfig().GetDynamicGpMode() != types.MinimalGpMode {
+		price := new(big.Int).Set(app.GlobalGp)
+		if price.Cmp((*big.Int)(api.gasPrice)) == -1 {
+			price.Set((*big.Int)(api.gasPrice))
+		}
+
+		if appconfig.GetOecConfig().GetDynamicGpCoefficient() > 0 {
+			coefficient := big.NewInt(int64(appconfig.GetOecConfig().GetDynamicGpCoefficient()))
+			gpRes := new(big.Int).Mul(price, coefficient)
+			avgGP = gpRes
```
Comment (zjg555543) [inline app/rpc/namespaces/eth/api.go]: simplified code

File: app/rpc/namespaces/eth/api.go (hunk 1 above)
Comment (LeoGuo621) [reply app/rpc/namespaces/eth/api.go:307]: avgGP is unchanged and does not need to be reset

File: app/rpc/namespaces/eth/api.go (hunk 3 above)
Comment (LeoGuo621) [reply app/rpc/namespaces/eth/api.go]: done

**Review by zjg555543 — DISMISSED** [review]

**Review by zjg555543 — DISMISSED** [review]

**Review by zjg555543 — APPROVED** [review]

**Pull request merged**